src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -899,6 +899,15 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
         return false;
     }
 
+    private makeMembershipStateKey(localUserId: string, localDeviceId: string): string {
+        const stateKey = `${localUserId}_${localDeviceId}`;
+        if (/^org\.matrix\.msc3779\b/.exec(this.room.getVersion())) {
+            return stateKey;
+        } else {
+            return `_${stateKey}`;
+        }
```

Comment:
Ah nice preparing for the owned state events is a smart idea.

Reply:
Just to give background on this: We never use the state key to actually compute the state events.
Which is somewhat broken. In theory a proper state event has to use a spec conform key.
With our implementation this is not checked (we could in theory send multiple member events for one user for the same device by using random state keys.)
This is what allows to only update the part where we compute the membership.

We probably want to check if the state key is to spec before parsing a state event.

Reply:
Added an issue here to track this: https://github.com/matrix-org/matrix-js-sdk/issues/4265